runelite-client/src/main/java/net/runelite/client/plugins/inventoryviewer/InventoryViewerOverlay.java

```diff
@@ -92,61 +124,100 @@ public Dimension render(Graphics2D graphics)
 	{
 		final ItemContainer itemContainer = client.getItemContainer(InventoryID.INVENTORY);
 
+		final Item[] items;
 		if (itemContainer == null)
 		{
-			return null;
+			items = new Item[0];
+		}
+		else
+		{
+			items = itemContainer.getItems();
 		}
 
-		panelComponent.getChildren().clear();
-
-		final Item[] items = itemContainer.getItems();
+		inventoryComponent.getChildren().clear();
+		wrapperComponent.getChildren().clear();
+		int wrapperWidth;
 
 		if (config.groupItems())
 		{
-			// Store in a LinkedHashMap to preserve inventory ordering
-			final Map<Integer, Integer> totals = Arrays.stream(items)
-				.filter(p -> p.getId() != -1)
-				.collect(groupingBy(Item::getId, LinkedHashMap::new, summingInt(Item::getQuantity)));
+			if (groupedItems == null)
+			{
+				// Store in a LinkedHashMap to preserve inventory ordering
+				groupedItems = Arrays.stream(items)
+					.filter(p -> p.getId() != -1)
+					.collect(groupingBy(Item::getId, LinkedHashMap::new, summingInt(Item::getQuantity)));
+			}
 
-			for (Map.Entry<Integer, Integer> cursor : totals.entrySet())
+			for (Map.Entry<Integer, Integer> cursor : groupedItems.entrySet())
 			{
 				final BufferedImage image = itemManager.getImage(cursor.getKey(), cursor.getValue(), true);
 				if (image != null)
 				{
-					panelComponent.getChildren().add(new ImageComponent(image));
+					inventoryComponent.getChildren().add(new ImageComponent(image));
 				}
 			}
 
 			// Add a placeholder if the inventory is empty, so the overlay can still be easily seen
-			if (totals.entrySet().size() == 0)
+			if (groupedItems.entrySet().size() == 0)
 			{
-				panelComponent.getChildren().add(inventoryIconSprite);
+				inventoryComponent.getChildren().add(inventoryIconSprite);
+				wrapperWidth = WRAPPER_MINIMUM_WIDTH;
+			}
+			else
+			{
+				final int widthItems = Math.min(INVENTORY_WIDTH, groupedItems.entrySet().size());
+				wrapperWidth = widthItems * Constants.ITEM_SPRITE_WIDTH + (widthItems + 1) * SPRITE_PADDING.x;
```

Comment:
This doesn't use the same formula as is used for the non-grouped inventory. What's below is:

```java
wrapperWidth = INVENTORY_WIDTH * Constants.ITEM_SPRITE_WIDTH + (INVENTORY_WIDTH - 1) * SPRITE_PADDING.x;
```

Specifically, **`(INVENTORY_WIDTH - 1)`**` * SPRITE_PADDING.x` compared to **`(widthItems + 1)`**` * SPRITE_PADDING.x`. Why are these different?